src/FTPClient.jl

```diff
@@ -16,6 +16,7 @@ type FTPClientError <: Exception
        lib_curl_error::UInt32
 end
 
+@enum FTP_MODE ascii_mode binary_mode
```

Comment:
This is kind of why I like the `baremodule` approach to enums:
```julia
baremodule FTP_MODE
    const binary = 1
    const ascii = 2
end
```

Then you do `FTP_MODE.binary` and `FTP_MODE.ascii` and just export `FTP_MODE`.

A couple years ago I tried to code this up as a macro but I was really bad at metaprogramming then. I should probably try again.